fvtest/porttest/omrsockTest.cpp

```diff
@@ -0,0 +1,150 @@
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "omrport.h"
+#include "omrportsock.h"
+#include "testHelpers.hpp"
+
+/**
+ * Start a server which includes create a socket, bind to an address/port and 
+ * listen for clients.
+ * 
+ * @param[in] portLibrary
+ * @param[in] testname The name of the calling test.
+ * @param[in] addrStr The address of the server.
+ * @param[in] port The server port.
+ * @param[in] family Socket address family wanted.
+ * @param[out] serverSocket A pointer to the server socket. 
+ * @param[out] serverAddr The socket address of the server created.
+ * 
+ * @return 0 on success, return an error otherwise.
+ */ 
+int32_t
+startServer(struct OMRPortLibrary *portLibrary, const char *testName, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *serverSocket, omrsock_sockaddr_t serverAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * Create the client socket then connect to the server.
+ *  
+ * @param[in] portLibrary
+ * @param[in] testname The name of the calling test.
+ * @param[in] addrStr The address of the server.
+ * @param[in] port The server port.
+ * @param[in] family Socket address family wanted.
+ * @param[in] sessionClientSocket A pointer to the client socket. 
+ * @param[in] sessionClientAddr The socket address of the client created. 
+ * 
+ * @return 0 on success, return an error otherwise.
+ */ 
+int32_t
+connectClientToServer(struct OMRPortLibrary *portLibrary, const char* testName, const char *addrStr, const char *port, int32_t family, omrsock_socket_t *sessionClientSocket, omrsock_sockaddr_t sessionClientAddr) 
+{
+    return OMRPORT_ERROR_NOTEXIST;
+}
+
+/**
+ * To test if Port Library is properly set up to run OMRSock tests. 
+ *
+ * This will iterate through all OMRSock functions in @ref omrsock.c and simply check 
+ * that the location of the function is not NULL.
+ * 
+ * @return TEST_PASS on success, TEST_FAIL on error.
+ *
+ * @note Errors such as an OMRSock function call with a NULL address will be reported.
+ */
+TEST(PortSockTest, sock_test0)
+{
+    OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+    const char *testName = "omrsock_test0";
+
+    reportTestEntry(OMRPORTLIB, testName);
+    reportTestExit(OMRPORTLIB, testName);
+}
+
+/**
+ * To test the omrsock per thread buffer.
+ * 
+ * In this test, per thread buffer related functions will be called to set up a buffer to store
```

Comment:
no need for future tense. otherwise, you will have to update the description after you add the implementation.
```suggestion
 * In this test, per thread buffer related functions set up a buffer to store
```